qiskit/dagcircuit/dagcircuit.py

```diff
@@ -388,6 +390,10 @@ def _add_op_node(self, op, qargs, cargs):
         new_node = DAGOpNode(op=op, qargs=qargs, cargs=cargs)
         node_index = self._multi_graph.add_node(new_node)
         new_node._node_id = node_index
+        if op.name in self._ops:
+            self._ops[op.name] += 1
+        else:
+            self._ops[op.name] = 1
```

Comment:
It might be less error prone and easier to read if we defined
```python
def _increment_op_count(op):
      if op.name in self._ops:
￼            self._ops[op.name] += 1
￼      else:
￼            self._ops[op.name] = 1  

def _decrement_op_count(op):
      if self._ops[op.name] == 1:
            del self._ops[op.name]
      else:
            self._ops[op.name] -= 1
 ```

Reply:
That's a good idea, done in: https://github.com/Qiskit/qiskit-terra/pull/6832/commits/da3804d43bc0ec9ecf65a82b158df09cea8a20dc